### PR TITLE
Use mitt as the event emitter 

### PR DESCRIPTION
**Pull request by OrKoN**
This PR replaces the outdated rollup polyfill[1] by a 200 byte event emitter called mitt[2] that comes with a wildcard event matching.

[1]: https://github.com/ionic-team/rollup-plugin-node-polyfills/issues/8
[2]: https://github.com/developit/mitt

### CODE REVIEW

File: src/utils/EventEmitter.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import mitt, {Emitter, EventType, Handler, WildcardHandler} from 'mitt';
+
+export class EventEmitter<Events extends Record<EventType, unknown>> {
```
Comment (sadym-chromium) [inline src/utils/EventEmitter.ts:20]: Why do we need this wrapper and cannot use mitt directly?

Comment (OrKoN) [reply src/utils/EventEmitter.ts:20]: to have a bit of control over the API and possible to instrument it in the future. E.g., in Puppeteer we have `removeAllListeners` implemented in the wrapper. But happy to remove the wrapper as it is strictly speaking is not needed.

File: src/utils/EventEmitter.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    type: Key,
+    handler: Handler<Events[Key]>
+  ): EventEmitter<Events>;
+  off(type: any, handler: any): EventEmitter<Events> {
+    this.#emitter.off(type, handler);
+    return this;
+  }
+
+  /**
+   * Emit an event and call any associated listeners.
```
Comment (thiagowfx) [inline src/utils/EventEmitter.ts:71]: Emits, use third person as per https://google.github.io/styleguide/jsguide.html#jsdoc-method-and-function-comments, ditto throughout

**Review by thiagowfx — APPROVED** [review]

File: src/utils/EventEmitter.ts (hunk 1 above)
Comment (sadym-chromium) [reply src/utils/EventEmitter.ts:20]: Agree, it's makes API more explicit having this wrapper

**Review by sadym-chromium — APPROVED** [review]

File: src/utils/EventEmitter.spec.ts [hunk 3]
```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright 2020 Google Inc. All rights reserved.
```
[the remark this answers is not in the document]
Comment (OrKoN) [reply src/utils/EventEmitter.spec.ts:2]: EventEmitter I wrote from scratch, the tests were adopted from Puppeteer's codebase, therefore, keeping the original license.

[the remark this answers is not in the document]
Comment (OrKoN) [reply src/utils/EventEmitter.spec.ts:2]: I am gonna land this and we can discuss the headers separately.

**Pull request merged**